src/helpers.c

```diff
@@ -123,6 +123,27 @@ size_t find_key(Key_File key_file, char *group, char *key) {
   return -1;
 }
 
+// Append a new key to an existing Key_File
+void new_key(Key_File *key_file, char *group, char *key) {
+  key_file_resize(key_file);
+  setGroup(key_file, key_file->length - 1, group);
+  setKey(key_file, key_file->length - 1, key);
+}
+
+// Set value for the given group, key combination. If the combination
+// does not exist it is created.
+void setKeyValue(void (*function) (Key_File*, size_t, void*), Key_File *kf, char *group, char *key, void *value) {
```

Comment:
It may be an alternative to always pass a string value, making the function pointer unnecessary.